installer/install.iss

```diff
@@ -2163,34 +2163,36 @@ begin
 
     // Git Credential Manager for Windows
     RdbGitCredentialManager[GCM_Classic]:=CreateRadioButton(GitCredentialManagerPage,'Git Credential Manager','(DEPRECATED) The <A HREF=https://github.com/Microsoft/Git-Credential-Manager-for-Windows>Git Credential Manager for Windows</A> handles credentials e.g.'+#13+'for Azure DevOps and GitHub (requires .NET framework v4.5.1 or later).',TabOrder,Top,Left);
 
     // No credential helper
     RdbGitCredentialManager[GCM_None]:=CreateRadioButton(GitCredentialManagerPage,'None','Do not use a credential helper.',TabOrder,Top,Left);
 
-    // Restore the settings chosen during a previous install, if .NET 4.5.1
-    // or later is available.
+    // Restore the settings chosen during a previous install
+    case ReplayChoice('Use Credential Manager','Core') of
+        'Disabled': RdbGitCredentialManager[GCM_None].Checked:=True;
+        'Enabled': RdbGitCredentialManager[GCM_Classic].Checked:=True;
+        'Core': RdbGitCredentialManager[GCM_Core].Checked:=True;
+    else
+        RdbGitCredentialManager[GCM_Core].Checked:=True;
+    end;
+    // Auto-upgrade GCM to GCM Core in version v2.29.0
+    if RdbGitCredentialManager[GCM_Classic].Checked and ((PreviousGitForWindowsVersion='') or IsUpgrade('2.29.0')) then begin
+        RdbGitCredentialManager[GCM_Core].Checked:=True;
+        AddToSet(CustomPagesWithUnseenOptions,GitCredentialManagerPage.ID);
+    end;
+
+    // Deactivate if each GCM doesn't meet their Fx version requirement.
+    if DetectNetFxVersion()<394254 then begin
+        RdbGitCredentialManager[GCM_Core].Checked:=False;
+        RdbGitCredentialManager[GCM_Core].Enabled:=False;
+    end;
     if DetectNetFxVersion()<378675 then begin
         RdbGitCredentialManager[GCM_Classic].Checked:=False;
         RdbGitCredentialManager[GCM_Classic].Enabled:=False;
-        RdbGitCredentialManager[GCM_Core].Checked:=False;
-        RdbGitCredentialManager[GCM_Core].Enabled:=False;
-    end else begin
-        case ReplayChoice('Use Credential Manager','Core') of
-            'Disabled': RdbGitCredentialManager[GCM_None].Checked:=True;
-            'Enabled': RdbGitCredentialManager[GCM_Classic].Checked:=True;
-            'Core': RdbGitCredentialManager[GCM_Core].Checked:=True;
-        else
-            RdbGitCredentialManager[GCM_Core].Checked:=True;
-        end;
-        // Auto-upgrade GCM to GCM Core in version v2.29.0
-        if RdbGitCredentialManager[GCM_Classic].Checked and ((PreviousGitForWindowsVersion='') or IsUpgrade('2.29.0')) then begin
```

Comment:
Why not keep this block here, but enhance the conditionals so that e.g. the auto-upgrade won't happen if `RdbGitCredentialManager[GCM_Core].Enabled` evaluated to `True`?

Reply:
My reason is that denesting the conditionals reduces complexity/ improves maintainability. The tradeoff is that the upgrade/replay execute regardless, and are the repealed by the version checks. but I think that is better than having a combinatory nested ifs.

If you look at structure of final code (rather than just diff) it is:

- Replay previous
- Upgrade to Core
- Version Guard for GCM Core
- Version Guard for GCM Classic


Did notice one thing that might need changed (based on testing): if both guards get triggered (and unchecked and disabled), do we need to manually set `[GCM_None].Checked:=True`, or do radios enforce that once option is chosen.

Reply:
My rationale: if you put the version guard _before_ replying, you can automatically reject an invalid replay.

Reply:
On that note, @y2kbugger, have you tested what happens if `GCM_Core` is checked during a replay, but then disabled because of the version check (using a bogus minimum version to force it to be disabled would be the easiest way to test that, I think)?